app/controllers/reservations_controller.rb

```diff
@@ -1,7 +1,26 @@
 class ReservationsController < ApplicationController
+  before_action :authenticate_user!, except: :create
+  before_action :set_reservation!, only: %i(destroy)
+
   def create
     @reservation = Reservation.new(item_id: params[:item_id])
     return unless @reservation.save
     redirect_to wishlists_for_guest_path(Wishlist.find(params[:wishlist_id]).token), notice: t(".success")
   end
+
+  def destroy
+    authorize! :destroy, @reservation
+    @reservation.destroy!
+    redirect_to wishlist_path(params[:wishlist_id]), notice: t(".success")
+  rescue CanCan::AccessDenied, ActiveRecord::RecordNotDestroyed
+    redirect_to wishlist_path(params[:wishlist_id]), alert: t("generic_error")
+  end
+
+  private
+
+  def set_reservation!
```

Comment:
What's more, there's only one method using this `before` callback, so why even use it. Just put the `@reservation = Reservation.find(params[:id])` back into the `destroy` as you would normally. When a need arises to deduplicate the code, then you'll use such callback. Right now, there is no need for it.